reader.go

```diff
@@ -60,10 +61,17 @@ func (r *Reader) BBox() Box {
 
 // Read and parse headers in the Shapefile. This will
 // fill out GeometryType, filelength and bbox.
-func (r *Reader) readHeaders() {
+func (r *Reader) readHeaders() error {
 	// don't trust the the filelength in the header
 	r.filelength, _ = r.shp.Seek(0, io.SeekEnd)
 
+	r.shp.Seek(0, 0)
+	var magic int32
+	binary.Read(r.shp, binary.BigEndian, &magic)
+	if magic != 9994 {
+		return errors.New("File signature doesn't match shapefile")
```

Comment:
Please do not capitalize error message (https://github.com/golang/go/wiki/CodeReviewComments#error-strings) and avoid contractions. (`don't` -> `do not`)
```suggestion
		return errors.New("file signature does not match shapefile")
```